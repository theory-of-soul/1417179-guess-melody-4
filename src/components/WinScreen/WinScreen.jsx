import React from "react";
import PropTypes from 'prop-types';

const WinScreen = (props) => {
  const {onClickReplayHandler} = props;
  return (
    <section className="result">
      <div className="result__logo"><img src="img/melody-logo.png" alt="Угадай мелодию" width="186" height="83"/></div>
      <h2 className="result__title">Вы настоящий меломан!</h2>
      <p className="result__total">Вы ответили правильно на 6 вопросов и совершили 2 ошибки</p>
      <button className="replay" type="button" onClick={onClickReplayHandler}>Сыграть ещё раз</button>
    </section>
  );
};

WinScreen.propTypes = {
  onClickReplayHandler: PropTypes.func.isRequired
};

export default WinScreen;
