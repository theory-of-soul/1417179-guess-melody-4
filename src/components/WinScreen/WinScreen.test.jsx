import React from "react";
import renderer from 'react-test-renderer';
import WinScreen from "./WinScreen";

describe(`WinScreen snapshot tests`, () => {
  it(`WinScreen should render 6 questions ant 2 errors`, () => {
    const component = renderer
      .create(<WinScreen onClickReplayHandler={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
