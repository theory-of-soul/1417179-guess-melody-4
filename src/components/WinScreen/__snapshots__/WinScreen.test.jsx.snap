// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinScreen snapshot tests WinScreen should render 6 questions ant 2 errors 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 6 вопросов и совершили 2 ошибки
  </p>
  <button
    className="replay"
    onClick={[Function]}
    type="button"
  >
    Сыграть ещё раз
  </button>
</section>
`;
